### ISSUE TITLE
Small typo

### ISSUE THREAD

**Issue by preet** [grisumbras/enum-flags#1]
Hi, I think there's a small typo on line 178 of flags.hpp:

```
flags &operator^=(enum_type e) noexcept {
        val_ &= static_cast<underlying_type>(e); // should be '^=' not '&='
        return *this;
    }
```

**Comment by grisumbras** [comment]
Indeed, thank you. I'm planning to work on this project on next weekend.

**Comment by grisumbras** [comment]
Fi91f369d7b094b48caa2318b3506210aca543742f

**Comment by grisumbras** [comment]
Fixed via 91f369d7b094b48caa2318b3506210aca543742f.
